.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,161 @@
+            python: 3.5
+          - modules_tool: modules-tcl-1.147
+            python: 3.7
+          - modules_tool: modules-3.2.10
+            python: 3.5
+          - modules_tool: modules-3.2.10
+            python: 3.7
+          - modules_tool: modules-4.1.4
+            python: 3.5
+          - modules_tool: modules-4.1.4
+            python: 3.7
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{matrix.python}}
+        architecture: x64
+
+    - name: install OS & Python packages
+      run: |
+        # for modules tool
+        sudo apt-get install lua5.1 liblua5.1-0-dev lua-filesystem lua-posix lua-filesystem tcl tcl-dev
+        sudo ln -s /usr/lib/x86_64-linux-gnu/lua/5.1/posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.1/posix.so
+        # for GitPython, python-hglib
+        sudo apt-get install git mercurial
+        # dep for GC3Pie
+        sudo apt-get install time
+        # Python packages
+        pip --version
+        pip install --upgrade pip
+        pip --version
+        pip install -r requirements.txt
+        # git config is required to make actual git commits (cfr. tests for GitRepository)
+        git config --global user.name "Travis CI"
+        git config --global user.email "travis@travis-ci.org"
+        git config --get-regexp 'user.*'
+
+    - name: install GitHub token (if available)
+      env:
+        # see https://github.com/<username>/easybuild-framework/settings/secrets
+        GITHUB_TOKEN: ${{secrets.TEST_GITHUB_TOKEN}}
+      run: |
+        if [ ! -z $GITHUB_TOKEN ]; then
+          if [ "x${{matrix.python}}" == 'x2.6' ];
+              then SET_KEYRING="keyring.set_keyring(keyring.backends.file.PlaintextKeyring())";
+              else SET_KEYRING="import keyrings; keyring.set_keyring(keyrings.alt.file.PlaintextKeyring())";
+          fi;
+          python -c "import keyring; $SET_KEYRING; keyring.set_password('github_token', 'easybuild_test', '$GITHUB_TOKEN')";
+        fi
+
+    - name: install modules tool
+      run: |
+          # avoid downloading modules tool sources into easybuild-framework dir
+          cd $HOME
+          export INSTALL_DEP=$GITHUB_WORKSPACE/easybuild/scripts/install_eb_dep.sh
+          # install Lmod
+          source $INSTALL_DEP ${{matrix.modules_tool}} $HOME
+          # changes in environment are not passed to other steps, so need to create files...
+          echo $MOD_INIT > mod_init
+          echo $PATH > path
+          if [ ! -z $MODULESHOME ]; then echo $MODULESHOME > moduleshome; fi
+
+    - name: check sources
+      run: |
+          # make sure there are no (top-level) "import setuptools" or "import pkg_resources" statements,
+          # since EasyBuild should not have a runtime requirement on setuptools
+          SETUPTOOLS_IMPORTS=$(egrep -RI '^(from|import)[ ]*pkg_resources|^(from|import)[ ]*setuptools' * || true)
```

Comment:
Are your sure you don't want to handle spaces before "from|import"?
And what files is * matching here?